core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -79,6 +81,9 @@
         Setting.floatSetting("cluster.routing.allocation.balance.threshold", 1.0f, 0.0f,
             Property.Dynamic, Property.NodeScope);
 
+    public static final Setting<Float> INDEX_SHARD_WEIGHT_MULTIPLIER_SETTING =
```

Comment:
should we require this to be an integer and only use floats internally? It think we should keep it simple and don't ask folks to give us floats where possible?